src/CMakeLists.txt

```diff
@@ -58,15 +58,29 @@ add_feature_info(futimens GIT_USE_FUTIMENS "futimens support")
 
 # qsort
 
+# old-style BSD qsort_r() has the 'thunk' parameter as the first argument of the
+# comparison function:
 check_prototype_definition(qsort_r
-	"void qsort_r(void *base, size_t nmemb, size_t size, void *thunk, int (*compar)(void *, const void *, const void *))"
+	"void (qsort_r)(void *base, size_t nmemb, size_t size, void *thunk, int (*compar)(void *, const void *, const void *))"
 	"" "stdlib.h" GIT_QSORT_R_BSD)
 
+# new-style BSD and GNU qsort_r() has the 'thunk' parameter as the last
```

Comment:
This is maybe "new style FreeBSD qsort_r" -- I don't know that other BSDs have picked it up?

Reply:
maybe use POSIX qsort_r?  (reference: https://www.austingroupbugs.net/view.php?id=900 )

Reply:
Neither NetBSD nor OpenBSD even have `qsort_r()` (and they don't have `qsort_s()` either), so technically it is more correct to use the term 'FreeBSD', or indeed just 'POSIX' here.

I guess the "old-style BSD" comment should then really be "old-style FreeBSD", but I'm inclined to leave the `GIT_QSORT_R_BSD` as it has always been.